pkg/controller/actuator.go

```diff
@@ -268,11 +265,27 @@ func (a *actuator) Delete(ctx context.Context, log logr.Logger, ex *extensionsv1
 	namespace := ex.GetNamespace()
 	log.Info("Component is being deleted", "component", "", "namespace", namespace)
 
+	var istioNamespace string
+
 	istioNamespace, _, err := a.findIstioNamespaceForExtension(ctx, ex)
-	// TODO how to handle missing Gateway objects during deletion (NotFoundErrors)?
-	if err != nil {
+	if client.IgnoreNotFound(err) != nil {
 		return err
 	}
+	if apierrors.IsNotFound(err) {
+		exState, err := getExtensionState(ex)
+		if err != nil {
+			return err
+		}
+		if exState.IstioNamespace == nil {
```

Comment:
If `Reconcile` is executed partially, then its possible to for example create the seed resources, but not set the extensions IstioNamespace.